src/vscode-dts/vscode.d.ts

```diff
@@ -19046,7 +19046,29 @@ declare module 'vscode' {
 		/**
 		 * The value of this reference. The `string | Uri | Location` types are used today, but this could expand in the future.
 		 */
-		readonly value: string | Uri | Location | unknown;
+		readonly value: string | Uri | Location | ChatReferenceBinaryData | unknown;
+	}
+
+	/**
+	 * Represents binary data associated with a chat reference.
+	 */
+	export class ChatReferenceBinaryData {
```

Comment:
Worth thinking about how we can defer actually sending the data until it is needed. See how `DataTransferFile` for example. We also allow `DataTransferFile` to be passed to a workspace edit which means we never need to transfer the data to extensions